packages/typescript/base.js

```diff
@@ -0,0 +1,439 @@
+    "@typescript-eslint/ban-ts-comment": [
+      "error",
+      {
+        "ts-expect-error": "allow-with-description",
+        "ts-ignore": "allow-with-description",
+        "ts-nocheck": true,
+        "ts-check": true,
+        minimumDescriptionLength: 10,
+      },
+    ],
+
+    // We're not concerned about ts-lint: comments as they're not used anywhere in our codebase
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-tslint-comment.md
+    "@typescript-eslint/ban-tslint-comment": "off",
+
+    // Bans specific types from being used, e.g.: {}
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-types.md
+    "@typescript-eslint/ban-types": "error",
+
+    // Ensures that literals on classes are exposed as fields to avoid additional overhead of creating a closure
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/class-literal-property-style.md
+    "@typescript-eslint/class-literal-property-style": ["error", "fields"],
+
+    // Enforces the usage of Record<K, V> type instead of { [key: K]: V }
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-indexed-object-style.md
+    "@typescript-eslint/consistent-indexed-object-style": ["error", "record"],
+
+    // Enforces consistent usage of type assertions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-assertions.md
+    "@typescript-eslint/consistent-type-assertions": [
+      "error",
+      {
+        assertionStyle: "as",
+        objectLiteralTypeAssertions: "allow-as-parameter",
+      },
+    ],
+
+    // Enforces consistent use of type definitions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-definitions.md
+    "@typescript-eslint/consistent-type-definitions": ["error", "type"],
+
+    // Enforces consistent usage of type exports
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-exports.md
+    "@typescript-eslint/consistent-type-exports": [
+      "error",
+      { fixMixedExportsWithInlineTypeSpecifier: true },
+    ],
+
+    // Enforces consistent usage of type imports
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-imports.md
+    "@typescript-eslint/consistent-type-imports": [
+      "error",
+      { prefer: "type-imports", disallowTypeAnnotations: true },
+    ],
+
+    // Enforces usage of return type on functions and methods
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-function-return-type.md
+    "@typescript-eslint/explicit-function-return-type": [
```

Comment:
Is there a way to automatically disable this for redux-saga function like you disabled param reassign for immer?

Reply:
Sure, adding an `.eslintrc` file in the directory where sagas live and adding `{ "rules": { "@typescript-eslint/explicit-function-return-type": "off" } }`